spec-tests/build.gradle

```diff
@@ -0,0 +1,90 @@
+import org.gradle.api.tasks.testing.logging.TestExceptionFormat
+import org.gradle.api.tasks.testing.logging.TestLogEvent
+
+plugins {
+    id "java"
+    id "checkstyle"
+}
+
+repositories {
+    mavenCentral()
+}
+
+description = "Candlepin Spec Tests"
+
+dependencies {
+    implementation project(":client")
+
+    implementation 'com.squareup.okhttp3:okhttp:4.9.3'
+    implementation 'org.slf4j:slf4j-api:1.7.33'
+    implementation 'org.junit.jupiter:junit-jupiter:5.8.1'
+
+    testImplementation 'org.junit.jupiter:junit-jupiter-api:5.8.2'
+    testImplementation 'org.junit.jupiter:junit-jupiter-engine:5.8.2'
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+
+    checkstyle "com.puppycrawl.tools:checkstyle:8.29"
+    checkstyle "com.github.sevntu-checkstyle:sevntu-checks:1.36.0"
+}
+
+test {
+// TODO find a better way to exclude spec tests from normal "./gradlew test" run
+    onlyIf {
+        project.hasProperty("runSpec")
```

Comment:
This doesn't work, because any test filtering/exclude/include is only applied to the current test task, not the subproject test tasks run after it. I don't see a clear way in gradle docs on how to not run subproject tasks when running the parent task, so I think the current way is the simplest thing that works.

Reply:
since the current solution breaks the ability to run spec tests from the IDE, here's an alternative that does what we want, without breaking running from IDE:
- remove the runSpec workaround from the subproject build.gradle
- add this workaround to the top project build.gradle(by replacing `:run` with `:test`): https://stackoverflow.com/questions/51903863/gradle-disable-automatic-subproject-execution-for-specific-task

Reply:
@wottop I am able to reproduce the issue when using our docker container in the same way jenkins does: `./docker/test -m -c "cp-test -uuu"`. Apparently, the `-uuu` is an attempt at running the unit tests 3 times (although it doesn't seem it ever  worked).
Another way to reproduce is to simply run the test command twice, without a clean inbetween: `./gradlew clean test; ./gradlew test`. For whatever reason (probably something to do with the fact that gradle is a daemon with caching mechanisms involved that I am not familiar with, so project properties aren't working as expected?).

I tried the workaround I mentioned before (add this to the top project build.gradle test task (by replacing :run with :test): https://stackoverflow.com/questions/51903863/gradle-disable-automatic-subproject-execution-for-specific-task) and it works without triggering this problem.